api/proto/ankaios.proto

```diff
@@ -130,29 +130,81 @@ enum AddCondition {
+    string additionalInfo = 1;
+    oneof ExecutionStateEnum {
+        AgentDisconnected agentDisconnected = 2;
+        Pending pending = 3;
+        Running running = 4;
+        Succeeded succeeded = 5;
+        Failed failed = 6;
+        NotScheduled notScheduled = 7;
+    }
+}
+
+/**
+* The exact state of the workload cannot be determined, e.g., because of a broken connection to the responsible agent.
+*/
+message AgentDisconnected {}
+
+/**
+* The workload is not scheduled to run at any agent. This is signalized with an empty agent in the workload specification.
+*/
+message NotScheduled {}
+
+/**
+* The workload is going to be started eventually.
+*/
+enum Pending {
+    PENDING_INITAL = 0; /// The workload specification has not yet being received by an agent
+    PENDING_WAITING_TO_START = 1; /// The start of the workload will be triggered once all its dependencies are met.
+    PENDING_STARTING = 2; /// Starting the workload was scheduled at the corresponding runtime.
+    PENDING_STARTING_FAILED = 8; /// The starting of the workload by the runtime failed.
+
+}
+
 /**
-* An enum type describing the workload execution state.
+* The workload is operational.
 */
-enum ExecutionState {
-    EXEC_UNKNOWN = 0; /// The exact state of the workload cannot be determined, e.g., because of a broken connection to the responsible agent.
-    EXEC_PENDING = 1; /// The workload has not being scheduled yet.
-    EXEC_WAITING_TO_START = 2; /// The start of the workload will be triggered once all its dependencies are met.
-    EXEC_STARTING = 3; /// Starting the workload was scheduled at the corresponding runtime.
-    EXEC_RUNNING = 4; /// The workload is operational.
-    EXEC_SUCCEEDED = 5; /// The workload has successfully finished operation.
-    EXEC_FAILED = 6; /// The workload has failed either during startup ot during operation
-    EXEC_WAITING_TO_STOP = 7; /// Thhe deletion of the workload will be triggered once no 'waiting to start', 'starting' or 'running' workload depending on it exists.
-    EXEC_STOPPING = 8; /// Stopping the workload was scheduled at the corresponding runtimeg.
-    EXEC_REMOVED = 10; /// The workload has been deleted.
+enum Running {
+    RUNNING_OK = 0; /// The workload is operational.
+    RUNNING_WAITING_TO_STOP = 1; /// Thhe deletion of the workload will be triggered once no 'pending' or 'running' workload depending on it exists.
+    RUNNING_STOPPING = 2; /// Stopping the workload was scheduled at the corresponding runtimeg.
+    RUNNING_DELETE_FAILED = 8; /// The deletion of the workload by the runtime failed.
+}
+
+/**
+* The workload has successfully finished operation.
+*/
+enum Succeeded {
+    SUCCEEDED_OK = 0; /// The workload has successfully finished operation.
+    SUCCEEDED_DELETE_FAILED = 8; /// The deletion of the workload by the runtime failed.
+}
+
+/**
+* The workload has failed or is in a degraded state.
+*/
+enum Failed {
+    FAILED_NOK = 0; /// The workload has failed during operation
```

Comment:
What about calling this `FAILED_EXEC_FAILED`? The "NOK" seems to me too general.

Reply:
Done.